{libratbag,openrazer-daemon,whipper}: fix typelib error

Artturin opened the pull request:
Fixes https://github.com/NixOS/nixpkgs/issues/316701
Fixes https://github.com/NixOS/nixpkgs/issues/310205
Fixes https://github.com/NixOS/nixpkgs/issues/312306

```
[user@system:~]$ ratbagctl info
Traceback (most recent call last):
  File "/nix/store/4x1mm4cm6a891yyl1a9ky08fxril160h-libratbag-0.17/bin/ratbagctl", line 36, in <module>
    from gi.repository import Gio, GLib, GObject
  File "/nix/store/czg030czd0sc7vmwjf71afhd9zskf5n2-python3-3.11.9-env/lib/python3.11/site-packages/gi/importer.py", line 133, in create_module
    raise ImportError('cannot import name %s, '
ImportError: cannot import name Gio, introspection typelib not found
```

## Description of changes

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

Artturin committed:
libratbag: fix `ratbagctl` typelib error

`ImportError: cannot import name Gio, introspection typelib not found`

Artturin committed:
openrazer-daemon: Fix typelib error

`ImportError: cannot import name GLib, introspection typelib not found`

eclairevoyant reviewed (commented):
Don't we have `wrapGAppsNoGuiHook` for exactly this use case?

Artturin committed:
openrazer-daemon: move deps to correct attributes

Artturin committed:
whipper: Fix typelib error

`ImportError: cannot import name GLib, introspection typelib not found`

Sorixelle commented:
[`targetcli`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/os-specific/linux/targetcli/default.nix) also seems to be affected, can we fix it there as well?

Artturin committed:
targetcli: fix typelib error

Add pygobject3,glib,gobject-introspection to not rely on something propagating them
